src/connection/mc_connection_man.erl

```diff
@@ -76,17 +132,24 @@ process_error(_, Doc) ->
   erlang:error({bad_query, Doc}).
 
 %% @private
--spec request_raw(port(), mc_worker_api:database(), mongo_protocol:message(), module()) ->
+-spec request_raw(any(), mc_worker_api:database(), mongo_protocol:message(), module()) ->
   ok | {non_neg_integer(), [map()]}.
 request_raw(Socket, Database, Request, NetModule) ->
   Timeout = mc_utils:get_timeout(),
   ok = set_opts(Socket, NetModule, false),
   {ok, _, _} = mc_worker_logic:make_request(Socket, NetModule, Database, Request),
   Responses = recv_all(Socket, Timeout, NetModule),
   ok = set_opts(Socket, NetModule, true),
-  {_, Reply} = hd(Responses),
+  Reply = hd(Responses),
   reply(Reply).
 
+%% @private
+request_raw_no_parse(Socket, Database, Request, NetModule) ->
+  Timeout = mc_utils:get_timeout(),
+  ok = set_opts(Socket, NetModule, false),
+  {ok, _, _} = mc_worker_logic:make_request(Socket, NetModule, Database, Request),
+  recv_all(Socket, Timeout, NetModule).
```

Comment:
😳😢 After setting back this setting there is no need to use a temporary connection to detect the protocol type. I though I did not get back responses from Mongo DB because it refused to do anything after getting an incorrect command but now I understand what the real reason was. My bad for not understanding the code good enough! Thanks!